### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by nsoranzo**
- Replace `invoke_workflow()` call with `invoke_workflow_and_assert_ok()`. Follow-up on commit 18891f3791b475e6c81569d9349a6fa8a8c62d50 .
- Replace use of deprecated API endpoint in `BaseWorkflowPopulator.wait_for_invocation()`.
- More type annotations for `CwlPopulator`.
- Fix error message for `RegexValidator`.
- Remove file added by mistake in commit 9690680ec7256d979b8b11eb7d5f343a691c93f3 .
- Remove `--wait` startup option. Follow-up on https://github.com/galaxyproject/galaxy/pull/13071 .

See individual commits for more details.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy_test/base/populators.py [hunk 1]
```diff
@@ -323,7 +323,7 @@ def _run_cwl_workflow_job(
             # workflows as well, and then make it the default. Or decide they are safe.
             "allow_tool_state_corrections": True,
         }
-        invocation_id = self.workflow_populator.invoke_workflow(
```
Comment (jmchilton) [inline lib/galaxy_test/base/populators.py:326]: Thanks for the fix, sorry about this.

Comment (nsoranzo) [reply lib/galaxy_test/base/populators.py:326]: No worries, this is tested only in the CWL branch.

File: lib/galaxy/tools/parameters/validation.py [hunk 2]
```diff
@@ -133,7 +133,7 @@ def from_element(cls, param, elem):
 
     def __init__(self, message, expression, negate):
         if message is None:
-            message = f"Value '%s' does {'not ' if negate == 'false' else ''}match regular expression '{expression}'"
+            message = f"Value '%s' does {'not ' if negate == 'false' else ''}match regular expression '{expression.replace('%', '%%')}'"
```
Comment (bernt-matthias) [inline lib/galaxy/tools/parameters/validation.py:136]: What is the problem solved here?

Comment (mvdbeek) [reply lib/galaxy/tools/parameters/validation.py:136]: It's in the commit message:
```
In [1]: expression = "5 %"
   ...: negate = False
   ...: value = 1
   ...: f"Value '%s' does {'not ' if negate == 'false' else ''}match regular expression '{expression}'" % value
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-1754553231d4> in <module>
      2 negate = False
      3 value = 1
----> 4 f"Value '%s' does {'not ' if negate == 'false' else ''}match regular expression '{expression}'" % value

TypeError: not enough arguments for format string
```
With the fix:
```
In [2]: f"Value '%s' does {'not ' if negate == 'false' else ''}match regular expression '{expression.replace('%', '%%')}'" % value
Out[2]: "Value '1' does match regular expression '5 %'"

```

**Review by mvdbeek — APPROVED** [review]

**Pull request merged**